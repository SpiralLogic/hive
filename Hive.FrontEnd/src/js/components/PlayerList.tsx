import * as React from 'react';
import { useHiveContext } from '../game-context';
import PlayerTiles from './PlayerTiles';

function PlayerList() {
    const { players } = useHiveContext();
    return (
        <div className="players">
            {players.map((player) => (
                <PlayerTiles key={player.id} {...player} />
            ))}
        </div>
    );
}

PlayerList.displayName = 'Player List';

export default PlayerList;
