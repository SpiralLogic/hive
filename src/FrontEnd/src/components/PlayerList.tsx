import { FunctionComponent, h } from 'preact';
import { PlayerId, Players } from '../domain';
import ActiveTile from './ActiveTile';
import Player from './Player';

const Players: FunctionComponent<{ players: Players; currentPlayer: PlayerId }> = (props) => {
  const { players, currentPlayer } = props;
  return (
    <aside className="players">
      {players.map((player) => (
        <Player name={player.name} id={player.id} currentPlayer={currentPlayer} hide={!!player.tiles.length}>
          {player.tiles.map((tile) => (
            <ActiveTile key={tile.id} {...tile} />
          ))}
        </Player>
      ))}
    </aside>
  );
};

Players.displayName = 'Player List';
export default Players;
