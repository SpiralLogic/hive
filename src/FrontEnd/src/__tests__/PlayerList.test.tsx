import { h } from 'preact';
import PlayerList from '../components/PlayerList';
import { renderElement } from './helpers';

const ant = { id: 1, playerId: 1, creature: 'ant', moves: [{ q: 1, r: 1 }] };
const fly = { id: 2, playerId: 0, creature: 'fly', moves: [] };

const players = [
    { id: 1, name: 'Player 1', tiles: [ant, fly, fly] },
    { id: 2, name: 'Player 2', tiles: [ant] },
];

const props = { players: players };
let playerList: HTMLElement;

beforeEach(() => {
    playerList = renderElement(<PlayerList {...props} />);
});

describe('Player List', () => {
    test('to have class', () => {
        expect(playerList).toHaveClass('players');
    });

    test('players are rendered', () => {
        expect(playerList.getElementsByClassName('player')).toHaveLength(2);
    });
});

describe('PlayerList snapshot', () => {
    test('matches current snapshot', () => {
        expect(playerList).toMatchSnapshot();
    });
});
