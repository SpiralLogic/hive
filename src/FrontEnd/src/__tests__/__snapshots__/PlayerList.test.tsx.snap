// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerList Tests PlayerList snapshot matches current snapshot 1`] = `
<aside
  class="players"
>
  <div
    class="player player1"
    title="Player 1"
  >
    <div
      class="name"
    >
      Player 1
    </div>
    <div
      class="tiles"
    >
      <div
        class="player1 hex tile"
        draggable="true"
        tabindex="0"
        title="ant"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#ant"
          />
        </svg>
      </div>
      <div
        class="player0 hex tile"
        draggable="false"
        title="fly"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#fly"
          />
        </svg>
      </div>
      <div
        class="player0 hex tile"
        draggable="false"
        title="fly"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#fly"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="player player2"
    title="Player 2"
  >
    <a
      class="name"
      href="/game/33/2"
      tabindex="-1"
    >
      Player 2
    </a>
    <div
      class="tiles"
    >
      <div
        class="player1 hex tile"
        draggable="true"
        tabindex="0"
        title="ant"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#ant"
          />
        </svg>
      </div>
    </div>
  </div>
</aside>
`;
